week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -21,8 +21,38 @@ This means the order the songs are played in will be random, but each song will
 
 class MusicPlayer {
     // Add your code here
+    constructor(){
+       this.index
+       this.music = []
+    }
+add(song, singer){
+let addSong = {
+    song: song,
+   singer: singer
+}
+this.music.push(addSong)
+}
+play(){
+    this.index = 0
+    if(this.music.length > 0 ){
+        console.log(`Currently playing: ${this.music[this.index].song} by ${this.music[this.index].singer}`)
+    } else {
+        console.log('Sorry there is no songs to play')
+    } 
 
 }
+skip(){
+this.index++
+
+if(this.index === this.music.length){this.index = 0}
+  console.log(`Currently playing: ${this.music[this.index].song} by ${this.music[this.index].singer}`);
+}
+previous(){
+    this.index--
+    if(this.index < 0){this.index = this.music.length - 1}
+    console.log(`Currently playing: ${this.music[this.index].song} by ${this.music[this.index].singer}`);
+}
+}
```

Comment:
Great job, just need to fix that small mistake on line 36. To simplify your code you could also call your `play()` function inside your `skip()` and `previous()` functions instead of typing out the console log again.

Well done Sulaiman!

Reply:
ok